src/lib.rs

```diff
@@ -0,0 +1,28 @@
+use std::net::SocketAddr;
+use clap::Parser;
+use hyper::Error;
+use routing::{get_main_router};
+use tracing::subscriber::SetGlobalDefaultError;
+use tracing_subscriber::FmtSubscriber;
+
+// declare child modules
+mod routing;
+
+#[derive(Debug, Parser)]
+pub struct LaunchConfig {
+    pub addr: SocketAddr
+}
+
+pub fn setup() -> Result<LaunchConfig, SetGlobalDefaultError> {
+    let subscriber = FmtSubscriber::new();
+    tracing::subscriber::set_global_default(subscriber)?;
```

Comment:
Tracing? Didn't expect to see this guy here, in such amateur project.

Reply:
We're using tokio anyway